### PR TITLE
test/meal-by-search-name

### PR DESCRIPTION
**Pull request by devnoorm7md**

### CODE REVIEW

File: src/main/kotlin/util/KMPMatcher.kt [hunk 1]
```diff
@@ -1,13 +1,29 @@
-package logic.usecase
+package util
 
-import logic.repository.FoodRepository
-import model.Food
-import model.NoMealsFoundException
+object KMPMatcher {
```
Comment (abdallahFc) [inline src/main/kotlin/util/KMPMatcher.kt]: This is not a utility class, it's logic ,move it to the logic package. Also, it doesn't need to be a singleton since I only use it in the search use case. There's no need for a single instance across the  application. It would be better to make it an abstraction by creating a search interface. This way we can implement different search algorithms. The use case should then depend on this interface

File: src/test/kotlin/util/KMPMatcherTest.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package util
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import kotlin.test.assertContentEquals
+
+class KMPMatcherTest {
```
Comment (abdallahFc) [inline src/test/kotlin/util/KMPMatcherTest.kt]: nice test for your search algorithm

File: src/main/kotlin/presentation/GetMealByNameUI.kt [hunk 3]
```diff
@@ -1,10 +1,10 @@
 package presentation
 
-import logic.usecase.GetMealBySearchForNameUseCase
+import logic.usecase.GetMealBySearchNameUseCase
 import presentation.ui_io.InputReader
 
 class GetMealByNameUI(
```
Comment (abdallahFc) [inline src/main/kotlin/presentation/GetMealByNameUI.kt:8]: write a test for this class also not depend on println fun depend on Printer interface

Comment (devnoorm7md) [reply src/main/kotlin/presentation/GetMealByNameUI.kt:8]: @AmrAshraf-git  will create a test for ui class

File: src/test/kotlin/presentation/GetMealByNameUITest.kt [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        // given
+        val input = null
+        every { reader.readString() } returns input
+
+        // when
+        ui.launchUI()
+
+        // then
+        verify {
+            printer.display("Enter the meal name to search: ")
```
Comment (eslam-99) [inline src/test/kotlin/presentation/GetMealByNameUITest.kt]: no need for verify such static lines "Enter the meal name to search: "

File: src/test/kotlin/logic/usecase/GetMealBySearchNameUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    fun `should throw NoMealsFoundException when repository returns empty list`() {
+        // given
+        every { repository.getFoods() } returns Result.success(emptyList())
+
+        // then
+        assertFailsWith<NoMealsFoundException> {
+            getMealBySearchNameUseCase.findMealsByName("Pizza")
+        }
+    }
+
```
Comment (nourelhodaahmed) [inline src/test/kotlin/logic/usecase/GetMealBySearchNameUseCaseTest.kt:187]: add a test case when all names in list is null should return NoMealsFoundException

Comment (devnoorm7md) [reply src/test/kotlin/logic/usecase/GetMealBySearchNameUseCaseTest.kt:187]: alreadya added

**Review by nourelhodaahmed — DISMISSED** [review]

**Review by eslam-99 — APPROVED** [review]

**Review by Noor-Serry — APPROVED** [review]

**Review by nourelhodaahmed — APPROVED** [review]

**Review by abdallahFc — APPROVED** [review]

**Review by Shahd-Hatem — APPROVED** [review]
